### DISCUSSION TITLE
Is it possible to use tree-sitter-hl-mode with flysplell-mode?

### DISCUSSION THREAD

**Discussion by Artawower** [emacs-tree-sitter/elisp-tree-sitter#168] [Q&A] [answered]
Hi! I'am trying use tree-sitter with enabled flyspell mode, and i can't see highlighting for words with mistakes
<img width="193" alt="image" src="https://user-images.githubusercontent.com/16963833/130352420-c25c1866-2b90-4e1b-92ec-b37492c99119.png">
and without tree-sitter-hl-mode
<img width="186" alt="image" src="https://user-images.githubusercontent.com/16963833/130352492-c67c9ee6-d969-4bdc-916c-f296bd0faeaa.png">
I found similar [issue](https://github.com/emacs-tree-sitter/elisp-tree-sitter/issues/64) but i don't know is it possible to use such solution for flyspell or not. 
When i use ispell-buffer i can see highlight on current selected word.

**Comment by ubolonton** [comment, marked as answer]
> I found similar [issue](https://github.com/emacs-tree-sitter/elisp-tree-sitter/issues/64) but i don't know is it possible to use such solution for flyspell or not.

Did that solution work for you?

I don't use `flyspell`, but when I try turning it on, it shows spelling mistakes *only when I navigate to the words*, regardless of whether `tree-sitter-hl` is on or not.

**Reply by Artawower** [reply]
No, this solution doesn't work for flyspell (I tried fu-spell and it worked nice, but fu-spell hasn't multiple language support)
I tried to find similar variable for describe flyspell`s fonts, 
```
  (setq-default flyspell-prog-text-faces
              '(tree-sitter-hl-face:comment
                tree-sitter-hl-face:doc
                tree-sitter-hl-face:string
                font-lock-comment-face
                font-lock-doc-face
                font-lock-string-face))
```
but without any luck.

https://user-images.githubusercontent.com/16963833/130866227-75a31bad-3bd0-4a33-9d83-1be025d0cbb5.mp4

<details>
<summary>My configs</summary>

```
  (setq ispell-program-name "aspell")
  ;; You could add extra option "--camel-case" for since Aspell 0.60.8
  ;; @see https://github.com/redguardtoo/emacs.d/issues/796
  (setq ispell-extra-args '("--sug-mode=ultra" "--lang=en_US" "--run-together" "--run-together-limit=16"))
  (setq-default flyspell-prog-text-faces
              '(tree-sitter-hl-face:comment
                tree-sitter-hl-face:doc
                tree-sitter-hl-face:string
                font-lock-comment-face
                font-lock-doc-face
                font-lock-string-face))
   (add-hook 'text-mode-hook 'flyspell-mode)
   (add-hook 'prog-mode-hook 'flyspell-prog-mode)

  (use-package tree-sitter
    :ensure t
    :after tree-sitter-langs
    :hook ((typescript-mode . tree-sitter-hl-mode)
           (js-mode . tree-sitter-hl-mode)
           (go-mode . tree-sitter-hl-mode)
           (python-mode . tree-sitter-hl-mode)
           (ng2-mode . tree-sitter-hl-mode))
    :config
    (push '(ng2-html-mode . html) tree-sitter-major-mode-language-alist)
    (push '(ng2-ts-mode . typescript) tree-sitter-major-mode-language-alist)
    (global-tree-sitter-mode)
    )

```
</details>

**Reply by ubolonton** [reply]
It looks like a bug in `flyspell` where `flyspell-generic-progmode-verify` assumes the `face` property is a single face, while it can actually be a list. I pushed a workaround in #170.

You would still need to set `flyspell-prog-text-faces`, though.

**Reply by Artawower** [reply]
Thanks so much! Now it works fine!
